### DISCUSSION TITLE
Non-static method Spatie\Analytics\Analytics::fetchVisitorsAndPageViews() cannot be called statically

### DISCUSSION THREAD

**Discussion by dutypro** [freekmurze/laravel-analytics#454] [General]
I've searched high and low on the internet but can't find an answer to this.

I have added Spatie Analytics but when I add this to my controller:

```
use Spatie\Analytics\Period;
use Spatie\Analytics\Analytics;

...
$analytics = Analytics::fetchVisitorsAndPageViews(Period::days(7));
...
```
I get the following error:

`Non-static method Spatie\Analytics\Analytics::fetchVisitorsAndPageViews() cannot be called statically`

Where am I going wrong? I can't find anything in the documentation about the error or how to rectify it.

**Comment by gmshawon** [comment]
Try to doing this:

use Analytics;

Instead of this:

use Spatie\Analytics\Analytics;

If that returns error saying "The use statement with non-compound name 'Analytics' has no effect", then no need to import ANALYTICS.  Try removing or commenting that line.

**Reply by dutypro** [reply]
`use Analytics`

Displays the error message: 

`Undefined type 'Analytics'.`

Strangely though, it still works.

**Reply by MarcusAfolabi** [reply]
> Try to doing this:
> 
> use Analytics;
> 
> Instead of this:
> 
> use Spatie\Analytics\Analytics;
> 
> If that returns error saying "The use statement with non-compound name 'Analytics' has no effect", then no need to import ANALYTICS. Try removing or commenting that line.

This works for me! Thanks

**Comment by Saifallak** [comment]
Try

```php
(new Spatie\Analytics\Analytics())->fetchVisitorsAndPageViews()
```

**Reply by hesham14yahia** [reply]
error expected 2 arguments found 0

**Reply by Saifallak** [reply]
Send parameters to the function.. what is the problem?

**Comment by Kankro** [comment]
For use this package first of you need to add into `config/app.php` the AnalyticsServiceProvider in the providers array.
```
'providers' => [
    ...
    Spatie\Analytics\AnalyticsServiceProvider::class,
]
```
And only after you can use 
```
use Analytics;
use Spatie\Analytics\Period;
 ...
$analyticsData = Analytics::fetchVisitorsAndPageViews(Period::days(7));
```

OR
you can use this method
```

use Spatie\Analytics\Analytics;
use Spatie\Analytics\AnalyticsClientFactory;
use Spatie\Analytics\Period;
 ...
$analyticsConfig = config('analytics');
$client = AnalyticsClientFactory::createForConfig($analyticsConfig);
$analytic = new Analytics($client, $analyticsConfig['view_id']);
$analyticsData = $analytic->fetchVisitorsAndPageViews(Period::days(7));
```
